axelrod/strategies/titfortat.py

```diff
@@ -665,3 +665,42 @@ def strategy(self, opponent: Player) -> Action:
         if opponent.history[-1] == D:
             return D
         return C
+
+@FinalTransformer((D,), name_prefix=None)
+class EugineNier(Player):
+    """
+    Plays similar to Tit-for-Tat, but with two conditions:
+    1) Always Defect on Last Move
+    2) If other player defects five times, switch to all defects.
+
+    Names:
+
+    - EugineNier Strategy: [LessWrong2011]_
+    """
+
+    name = 'EugineNier'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': {'length'},
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.is_defector = False
+
+    def strategy(self, opponent: Player) -> Action:
+        if not self.history:
+            return C
+        if self.is_defector or opponent.history[-5:] == [D] * 5:
```

Comment:
Please use `opponent.defections` instead of `opponent.history.count(D)`, it's more efficientPlease use `opponent.defections` instead of `opponent.history.count(D)`, it's more efficient